### ISSUE TITLE
[rule-proposal] no-middleman-promise

### ISSUE THREAD

**Issue by yfdyh000** [eslint/eslint#8899]
**Please describe what the rule should do:**
Eliminate intermediate variables for Promise, if the variable for the promise only.

**What category of rule is this? (place an "X" next to just one item)**

[?] Enforces code style
[ ] Warns about a potential error
[X] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
ref: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/StorageArea/clear
```js
var clearStorage = browser.storage.local.clear();
clearStorage.then(onCleared, onError);
```
to
```js
browser.storage.local.clear().then(onCleared, onError);
```
**Why should this rule be included in ESLint (instead of a plugin)?**
I guess it applies to all contexts.

https://github.com/mdn/webextensions-examples/issues/242

**Comment by platinumazure** [comment]
Feels subjective to me (i.e., developers might be declaring the variable for readability). Not sure it's critical to ESLint core.

**Comment by yfdyh000** [comment]
Yes, but it was a good discussion for me. Owner can do warn, not fixes only.

Maybe it fits as a plugin, I have not figured out the difference between it and the rules.

**Comment by yfdyh000** [comment]
Maybe it fits to https://github.com/xjamundx/eslint-plugin-promise, @xjamundx.
```
 * Rule: prefer-simple-then
 * Discourage using intermediate variables to store promise for then()
```

**Comment by yfdyh000** [comment]
If want to keep variables name as semantics, can do an option for adding an line comment with variable name (and fixed / customizable prefix).

Anyway, it does not seem to be favored as a core rule.
